### PR TITLE
Replace all hard coded localhosts with env constant variables

### LINKED ISSUE #1244: App can only be accessed on `localhost`
**Describe the bug**
1. The codebase hard codes localhost in many places
1. This means you cannot get Talawa-Admin to work while being accessed from a remote host without a proxy

**To Reproduce**

1. Install  Talawa-Admin on a remote system
1. Errors are seen when accessing the app directly from a remote host

**Expected behavior**
1. Access must be available from a remote host that is not located on `localhost`. This can only be truly tested when run from a completely different system.
1. The installation guide must show how to access Admin from a remote system
1. All hard coded instances of `localhost` must be replaced by configuration values in `.env`  and/or the browser URL where applicable

**Actual behavior**
- See screenshots

**Screenshots**

1. The `.env` file
image
1. Devtools output
image
1. The codebase hard codes localhost in many places
image

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by NayOoLwin5**
**What kind of change does this PR introduce?**

Replacement of all hard coded localhost urls in places with environment constant variables.

**Issue Number:**

Fixes #1244 

**Did you add tests for your changes?**

**Snapshots/Videos:**
Screenshot 2024-01-25 at 20 23 40

**If relevant, did you update the documentation?**

Yes

**Summary**
- With PR #1488 merged, I had to introduce new env variable `REACT_ADMIN_FRONTEND_HOST` for two test files to access host IP and port separately and wrote a prompt for it in setup
- Leaving hard coded localhost in file `/src/components/AddOn/support/services/Plugin.helper.ts` as I was informed it is outdated and no longer used

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@NayOoLwin5 Please fix the test application failing tests.

File: src/screens/LoginPage/LoginPage.test.tsx [hunk 1]
```diff
@@ -144,7 +145,7 @@ describe('Talawa-API server fetch check', () => {
       );
     });
 
-    expect(fetch).toHaveBeenCalledWith('http://localhost:4000/graphql/');
+    expect(fetch).toHaveBeenCalledWith(BACKEND_URL);
```
Comment (palisadoes) [inline src/screens/LoginPage/LoginPage.test.tsx:148]: Why is this necessary for testing when we will always be testing against localhost?

File: src/screens/UserPortal/Home/Home.test.tsx [hunk 2]
```diff
@@ -215,7 +219,9 @@ async function wait(ms = 100): Promise<void> {
 }
 
 beforeEach(() => {
-  const url = `http://localhost:${process.env.PORT}/user/organization/id=orgId`;
+  const url = REACT_APP_CUSTOM_PORT
```
Comment (palisadoes) [inline src/screens/UserPortal/Home/Home.test.tsx:222]: Why can't this be a relative URL?

File: src/screens/EventDashboard/EventDashboard.test.tsx [hunk 3]
```diff
@@ -44,10 +48,13 @@ async function wait(ms = 500): Promise<void> {
 
 describe('Testing Event Dashboard Screen', () => {
   beforeEach(() => {
+    const url = REACT_APP_CUSTOM_PORT
+      ? `${REACT_ADMIN_FRONTEND_HOST}:${REACT_APP_CUSTOM_PORT}/event/event123`
```
Comment (palisadoes) [inline src/screens/EventDashboard/EventDashboard.test.tsx:52]: Why can't this be a relative URL?

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. I don't see the need to create the `REACT_ADMIN_FRONTEND_HOST` variable when relative variables would seem to be equally suitable.
2. Let us know why this is necessary
3. The replacement of `localhost` with the `BACKEND_URL` is clearly necessary.

**Comment by NayOoLwin5** [comment]
- I checked and found out all test files have its mock responses against the localhost. Therefore, no need of dynamic env variables. Thanks for pointing them out.
- However, running tests only with relative url resulting in failures. Thus, I kept them same as it was. Please review and let me know if anything more needed to be fixed. 

@palisadoes

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by NayOoLwin5** [comment]
- I cannot reproduce the test failures on my machine even though using node version 20. 
- However, I tested the login page manually by myself and the fetch checks are working as expected with my new changes. 
- No changes to the failed test files also as I changed back to previously existing code. 

Please let me know how I can reproduce the failed tests. @palisadoes

**Comment by palisadoes** [comment]
1. Scroll down in the test box
2. Click on this link
3. It is explained there

image

**Comment by NayOoLwin5** [comment]
Hi @palisadoes, the test should work fine now. The issue was removing dynamic env `BACKEND_URL` variable from the test files as the value in fetch expect the same as that is parsed in fetch in the components. Added it back and let us check the test again.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**